metric_learn/itml.py

```diff
@@ -170,13 +178,21 @@ def fit(self, pairs, y, bounds=None):
         If not provided at initialization, bounds_[0] and bounds_[1] will be
         set to the 5th and 95th percentile of the pairwise distances among all
         points present in the input `pairs`.
+    threshold_params : `dict` or `None`
+        Dictionary of parameters to give to `calibrate_threshold` for the
+        threshold calibration step done at the end of `fit`. If `None` is
+        given, `calibrate_threshold` will use the default parameters.
 
     Returns
     -------
     self : object
         Returns the instance.
     """
-    return self._fit(pairs, y, bounds=bounds)
+    self._fit(pairs, y, bounds=bounds)
+    self.calibrate_threshold(pairs, y, **(threshold_params if
```

Comment:
For all algorithms we should test that `threshold_params` has allowed values at the beginning of `fit` otherwise we will fit for nothing (which may take some time) if bad params have been given...